spec-docs/annotation-options.md

```diff
@@ -0,0 +1,169 @@
+
+``` kotlin
+package kotlin
+
+enum class AnnotationTarget {
+    PACKAGE
+    ...
+}
+
+target(ANNOTATION_CLASS)
+annotation(RUNTIME) class target(vararg targets: AnnotationTarget)
+```
+
+When loading an annotation, we only read `kotlin.target`. When `kotlin.target` is missing, on the JVM, we read `j.l.a.Target` and map its values to Kotlin ones according to the table above. This implies that we can load pure Java annotations that know nothing about Kotlin, and that an annotation written in Java can be targeted, e.g. for Kotlin expressions, because one can simply manually specify `kotlin.target` for it.
+
+### Syntax
+
+It makes sense to use `kotlin.target` explicitly in Kotlin code:
+
+``` kotlin
+target(EXPRESSION, TYPE)
+annotation class MyAnn
+```
+
+> An alternative would be to make target a aproperty of `kotlin.annotation`, but then we'd
+* lose the advantage of varargs, because there are more optional parameters
+* be non-uniform with Java, thus making it harder to figure how to make a Java annotation Kotlin-friendly
+
+## Retention
+
+> NOTE: Retention is a Java-specific concern, more or less. CLR retains all attributes at runtime, and JS too
+
+It makes a lot of sense to make `RUNTIME` the default retention.
+
+Since `RetentionPolicy.CLASS` is not a good fit for Kotlin that has functions outside any class, it's better to have `BINARY` instead. Also, we could have use `java.lang.annotation.RetentionPolicy` anyways, since it's platform-specific. Thus, we need to have our own enum:
+
+``` kotlin
+package kotlin
+
+enum class AnnotationRetention {
+    SOURCE
+    BINARY
+    RUNTIME
+}
+```
+
+> Now, we could map `java.lang.annotation.Retention` and `RetentionPolicy` to `kotlin.retention` and `kotlin.AnnotationRetention`, and then map `CLASS` to `BINARY`, but that is a little too much
+
+Then, it makes sense to make `retention` a property of `kotlin.annotation`:
+
+``` kotlin
+target(TYPE)
+annotation(SOURCE) class MyAnn
+```
+
+The following checks must be performed at compile time:
+* `EXPRESSION`-targeted annotations can only have retention `SOURCE`
+
+## Repeatable
+
+> Java has `Repeatable` as an annotation, but we cannot map a Kotlin type to it, because it is only present since JDK 8, and cannot be written to class files with version lower than 8.
+
+We make `repeatable` a boolean property of `kotlin.annotation`, with default value `false` (as in Java and C#).
+
+If a non-repeatable annotation is used multiple times on the same element, it is a compile-time error.
+
+If a repeatable annotation is used multiple times on the same element, but the target byte code version is lower than Java 8, it is a compile-time error.
+
+## Inherited and Documented
+
+These two options have ratehr unclear value, and we do not supprt them in Kotlin. One can use platform-specific annotations to express them.
+
+## Appendix. Definition of kotlin.annotation
+
+``` kotlin
+package kotlin
+
+target(CLASSIFIER)
```

Comment:
Well, it's not an annotation class until this annotation is applied to it... But maybe you are right, not sure